projects/packages/forms/src/contact-form/class-contact-form.php

```diff
@@ -384,24 +384,54 @@ public static function get_default_to( $post_author_id = null ) {
 		return $default_to;
 	}
 
+	/**
+	 * Safely get a property from post data (object or array).
+	 *
+	 * @param mixed  $post_data Post data (object or array).
+	 * @param string $property  Property name to get.
+	 *
+	 * @return mixed|null The property value or null if not found.
+	 */
+	public static function get_post_property( $post_data, $property ) {
+		if ( ! $post_data ) {
+			return null;
+		}
+
+		if ( is_object( $post_data ) && isset( $post_data->$property ) ) {
+			return $post_data->$property;
+		} elseif ( is_array( $post_data ) && isset( $post_data[ $property ] ) ) {
+			return $post_data[ $property ];
+		}
+
+		return null;
+	}
+
 	/**
 	 * Get the default subject for the contact form.
 	 *
 	 * @param array $attributes The attributes of the contact form.
+	 * @param mixed $post_data Optional post data (object or array).
 	 *
 	 * @return string The default subject for the contact form.
 	 */
-	public static function get_default_subject( $attributes ) {
+	public static function get_default_subject( $attributes, $post_data = null ) {
```

Comment:
There are other references to get_default_subject that we did not update. I assume in those cases, we're fine having $post_data = null. It just means that the first effort to set $post_title below will result in null, so we'll try to set the $post_title from the $post global variable.